src/main/java/com/scalar/cassy/server/RemoteCommandHandler.java

```diff
@@ -2,33 +2,38 @@
 
 import com.google.common.util.concurrent.Uninterruptibles;
 import com.google.inject.Inject;
-import com.scalar.cassy.db.DatabaseAccessor;
+import com.scalar.cassy.config.CassyServerConfig;
+import com.scalar.cassy.db.BackupHistory;
+import com.scalar.cassy.db.RestoreHistory;
 import com.scalar.cassy.exception.RemoteExecutionException;
 import com.scalar.cassy.remotecommand.RemoteCommandContext;
 import com.scalar.cassy.remotecommand.RemoteCommandResult;
 import com.scalar.cassy.rpc.OperationStatus;
 import com.scalar.cassy.service.BackupServiceMaster;
+import com.scalar.cassy.util.ConnectionUtil;
 import java.io.IOException;
+import java.sql.Connection;
 import java.util.concurrent.BlockingQueue;
 import java.util.concurrent.TimeUnit;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class RemoteCommandHandler implements Runnable {
   private static final Logger logger = LoggerFactory.getLogger(RemoteCommandHandler.class);
+  private final CassyServerConfig config;
   private final BlockingQueue<RemoteCommandContext> futures;
-  private final DatabaseAccessor database;
+  private boolean isActive = true;
```

Comment:
I think we should add `volatile` modifier or use `AtomicBoolean` instead of `boolean` here as this `isActive` field is accessed by multiple threads.

Reply:
Good catch! Thank you!
I'll add a `volatile` keyword.
BTW, using `AtomicBoolean` fixes the cache issue?
(if `AtomicBoolean` is used, a written value by a thread will be guaranteed to be read by another thread immediatelyly?)

Reply:
Yes, using `AtomicBoolean` should fix the cache issue:
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/package-summary.html

Reply:
I didn't know that. Thank you!